### DISCUSSION TITLE
How to set a default background image for articles?

### DISCUSSION THREAD

**Discussion by goldsheep3** [nunocoracao/blowfish#1022] [Q&A]
*I'm sorry, I'm from China, and my English is not very good, so I used ChatGPT to help with the translation of this content. Please understand.*

In the recent past, I planned to migrate my blog to Hugo and selected the Blowfish theme. During the process of migrating the blog to Blowfish, I was trying to set the background image for my articles.

For a specific article, I can set the `FeaturedImage` and also set `showHero` to true with `heroStyle` as "background" to display the background image. However, for all articles, I can set a `defaultFeaturedImage`. But when I set `showHero` to true and `heroStyle` to "background" in this case, the article does not display the background image, and the default FeaturedImage is not set as the article's HeroImage.

I want to avoid creating all articles as folder structures and placing the same `featured.jpg` in each folder to achieve the result of setting a background image for all articles. Is there a way to solve this issue?

**Comment by nunocoracao** [comment]
Hello @goldsheep3 

Use the `defaultBackgroundImage` in params.toml

```
defaultBackgroundImage = "IMAGE.jpg" # used as default for background images 
```

more info here - https://blowfish.page/docs/configuration/#global-1

![Screenshot 2023-12-01 at 11 39 55](https://github.com/nunocoracao/blowfish/assets/3803196/6151572f-b40d-4949-b930-d56d085cbe93)

**Comment by jcarnat** [comment]
Hello,

Thanks to this answer, I could have all my posts have a default featured image.
But can I have a post overwrite this default and get its own particular featured image?

I have tried all of those in Front Matter but none worked :(
```
background: "/img/pufferfish.png"
featuredImage: "/img/pufferfish.png"
FeaturedImage: "/img/pufferfish.png"
featured: "/img/pufferfish.png"
cover: "/img/pufferfish.png"
thumbnail: "/img/pufferfish.png"
```

Thanks.

**Reply by GrassBlock1** [reply]
> Hello,
> 
> Thanks to this answer, I could have all my posts have a default featured image.
> But can I have a post overwrite this default and get its own particular featured image?
> 
> I have tried all of those in Front Matter but none worked :(
> ```
> background: "/img/pufferfish.png"
> featuredImage: "/img/pufferfish.png"
> FeaturedImage: "/img/pufferfish.png"
> featured: "/img/pufferfish.png"
> cover: "/img/pufferfish.png"
> thumbnail: "/img/pufferfish.png"
> ```
> 
> Thanks.
> 
Hi @jcarnat ,
You need to put your featured image with your markdown file together in a folder.
Docs here: https://blowfish.page/samples/thumbnail_sample/

Or with Blowfish v2.55.0+ you can use featureimage front matter for a external featured image as URL. :D

**Reply by jcarnat** [reply]
Oh, so those are the only two options. I read the docs at least 10 times :) I thought the Front Matter option was not URL only.

Thanks!

**Comment by bitSheriff** [comment]
Is it possible to set a file (in the assets folder or static) as a feature image? This way, I would not have duplicated files in the folders of the posts if I would use the same featureImage more than once
